packages/react-async/src/tests/component.test.tsx

```diff
@@ -42,16 +42,27 @@ function Loading() {
   return <>Loading…</>;
 }
 
+jest.mock('@shopify/async', () => {
+  return {
+    ...jest.requireActual('@shopify/async'),
+    createResolver: jest.fn(),
+  };
+});
+
+const mockCreateResolver = jest.requireMock('@shopify/async').createResolver;
```

Comment:
I really don't like how I'm handling this mock and would be VERY open to suggestions here.

I couldn't figure out another way of handling this.

Reply:
I think the cleanest way would be what we chatted about before and break these out into another test suite 🤷

Reply:
yeah. Done